packages/core/src/application.ts

```diff
@@ -234,16 +237,28 @@ export class Application extends Context {
   ) {
     options = options || {version: '2.0', format: 'json'};
     let specObj = this.getApiSpec();
+    let appPort = await this.get('http.port');
+    // TODO make this more robust i.e. get host, protocol, etc. dynamically
+    let appUrl = 'http://127.0.0.1:' + appPort + '/swagger';
```

Comment:
The hostname and port should be default the `Host` header of the http request. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host

Reply:
Unfortunately, this will not work in production(like) environments, where the app is running behind a load balancer (nginx, Amalgam8, etc.). The `Host` header of the http request will be showing the internal hostname not accessible from outside. We could use `X-Forwarded-*` family of headers to determine the real public host, but I think it's not worth the effort.